packages/eslint-plugin/src/rules/no-misused-promises.ts

```diff
@@ -568,11 +571,26 @@ export default createRule<Options, MessageId>({
           continue;
         }
 
+        if (
+          node.type !== AST_NODE_TYPES.MethodDefinition &&
+          node.type !== AST_NODE_TYPES.TSAbstractMethodDefinition &&
+          node.type !== AST_NODE_TYPES.TSMethodSignature &&
+          node.type !== AST_NODE_TYPES.TSPropertySignature
```

Comment:
This is necessary mainly for type-checking, and I don't know if this can realistically have a runtime scenario. Please let me know if this should be handled differently (currently, there is no coverage for this).